File: week-04/yc_5430_ac.cs
using System;
using System.Collections.Generic;
using System.Text;

class Program
{
    static void Main(string[] args)
    {
        var testCase = int.Parse(Console.ReadLine());
        var sb = new StringBuilder();

        while(testCase > 0)
        {
            string commands = Console.ReadLine();
            var n = int.Parse(Console.ReadLine());
            string arrStr = Console.ReadLine();

            arrStr = arrStr.Substring(1, arrStr.Length - 2);
            var dq = new LinkedList<int>();
            if(!string.IsNullOrEmpty(arrStr))
            {
                foreach(var s in arrStr.Split(","))
                {
                    dq.AddLast(int.Parse(s));
                }
            }
            bool isForwardDirection = true;
            bool error = false;

            foreach(char c in commands)
            {
                if(c == 'R')
                {
                    isForwardDirection = !isForwardDirection;
                }
                else if(c == 'D')
                {
                    if(dq.Count == 0)
                    {
                        error = true;
                        break;
                    }
                    if(!isForwardDirection)
                    {
                        dq.RemoveLast();
                    }
                    else
                    {
                        dq.RemoveFirst();
                    }
                }
            }

            if(error)
            {
                sb.AppendLine("error");
            }
            else
            {
                sb.Append("[");
                if(dq.Count > 0)
                {
                    if(!isForwardDirection)
                    {
                        var node = dq.Last;
                        while(node != null)
                        {
                            sb.Append(node.Value);
                            node = node.Previous;
                            if(node != null)
                            {
                                sb.Append(",");
                            }
                        }
                    }
                    else
                    {
                        var node = dq.First;
                        while(node != null)
                        {
                            sb.Append(node.Value);
                            node = node.Next;
                            if(node != null)
                            {
                                sb.Append(",");
                            }
                        }
                    }
                }
                sb.AppendLine("]");
            }
            testCase--;
        }
        Console.Write(sb.ToString());
    }
}

File: week-04/yc_5430_ac.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuffer sb = new StringBuffer();

        int testCase = Integer.parseInt(br.readLine());

        while(testCase > 0) {
            String commands = br.readLine();
            int n = Integer.parseInt(br.readLine());
            String arrStr = br.readLine();

            arrStr = arrStr.substring(1, arrStr.length() -1);
            Deque<Integer> dq = new ArrayDeque<>();
            if(!arrStr.isEmpty()) {
                String[] parts = arrStr.split(",");
                for(String p : parts) dq.addLast(Integer.parseInt(p));
            }
            boolean isForwardDirection = true;
            boolean isError = false;

            for(char c: commands.toCharArray()) {
                if(c == 'R') {
                    isForwardDirection = !isForwardDirection;
                } else if(c == 'D') {
                    if(dq.isEmpty()) {
                        isError = true;
                        break;
                    }
                    if(!isForwardDirection) {
                        dq.removeLast();
                    }
                    else {
                        dq.removeFirst();
                    }
                }
            }

            if (isError) {
                sb.append("error\n");
            }
            else {
                sb.append("[");
                if(!dq.isEmpty()) {
                    if(!isForwardDirection) {
                        Iterator<Integer> it = dq.descendingIterator();
                        while(it.hasNext()) {
                            sb.append(it.next());
                            if(it.hasNext()) {
                                sb.append(",");
                            }
                        }
                    }
                    else {
                        Iterator<Integer> it = dq.iterator();
                        while(it.hasNext()) {
                            sb.append(it.next());
                            if(it.hasNext()) {
                                sb.append(",");
                            }
                        }
                    }
                }
                sb.append("]\n");
            }
            testCase--;
        }
        System.out.println(sb.toString());
    }
}